spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/AbstractMessageListenerContainer.java

```diff
@@ -327,13 +327,16 @@ private QueueAttributes queueAttributes(String queue, SqsMessageDeletionPolicy d
 			return null;
 		}
 
-		GetQueueAttributesResult queueAttributes = getAmazonSqs().getQueueAttributes(
-				new GetQueueAttributesRequest(destinationUrl).withAttributeNames(QueueAttributeName.RedrivePolicy));
+		GetQueueAttributesResult queueAttributes = getAmazonSqs()
```

Comment:
There is a small problem here. Calling `GetQueueAttributes` with attribute `FifoQueue` on a non-fifo queue results in error `Unknown Attribute FifoQueue`. 

Another way to determine if a queue is a fifo queue, is to check if it's name ends with ".fifo" (its not possible to create a fifo queue that does not have ".fifo" suffix and it's not possible to create a queue with ".fifo" suffix that is not a FIFO queue.

Reply:
I have now changed it to only check the suffix, but I still kept the additional fifo-property in QueueAttributes. Would you suggest to remove this one too?

Reply:
Thanks! Lets keep it there.